packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -679,7 +691,7 @@ export class DefaultWeb3Handler
       throw new Error('Non-EOA transaction detected')
     }
     // TODO: Make sure we lock this function with this nonce so we don't send to txs with the same nonce
-    const nonce = (await this.context.wallet.getTransactionCount()) + 1
+    const nonce = this.walletNonce + 1
```

Comment:
Ah one very annoying thing is I think that this approach will break tests that use snapshot... hmm

Reply:
Indeed it does, haha. Thanks for the suggested fix!